packages/core/src/actions/transactions/watchPendingTransactions.ts

```diff
@@ -0,0 +1,60 @@
+import type { Transaction } from 'ethers'
+import shallow from 'zustand/shallow'
+
+import { getClient } from '../../client'
+import type { Provider } from '../../types'
+import { debounce } from '../../utils'
+import { getProvider, getWebSocketProvider } from '../providers'
+
+export type WatchPendingTransactionsResult = Transaction
+export type WatchPendingTransactionsArgs = { chainId?: number }
+export type WatchPendingTransactionsCallback = (
+  transaction: WatchPendingTransactionsResult,
+) => void
+
+export function watchPendingTransactions(
+  args: WatchPendingTransactionsArgs,
+  callback: WatchPendingTransactionsCallback,
+) {
+  // We need to debounce the listener as we want to opt-out
+  // of the behavior where ethers emits a "block" event for
+  // every block that was missed in between the `pollingInterval`.
+  // We are setting a wait time of 1 as emitting an event in
+  // ethers takes ~0.1ms.
```

Comment:
No, wasn't sure about debouncing. I figured it might be a way to 'smooth' callbacks.. but in this case it might just 'lose' transactions? Will remove it

Reply:
Removed the debounce workaround and added docs. Should be good to merge